freeipa/src/main/java/com/sequenceiq/freeipa/service/AltusMachineUserService.java

```diff
@@ -208,11 +210,22 @@ private MonitoringCredential storeMonitoringCredential(AltusCredential altusCred
         monitoringCredential.setMachineUserName(getMonitoringMachineUser(stack));
         monitoringCredential.setAccessKey(altusCredential.getAccessKey());
         monitoringCredential.setPrivateKey(altusCredential.getPrivateKey() != null ? new String(altusCredential.getPrivateKey()) : null);
-        monitoringCredential.setAccessKeyType(cdpAccessKeyType.getValue());
-        String monitoringCredentialJsonString = new Json(monitoringCredential).getValue();
-        stack.setMonitoringCredential(monitoringCredentialJsonString);
-        stackService.save(stack);
-        return monitoringCredential;
+        return storeCdpCredential(monitoringCredential, stack.getId(), Stack::setMonitoringCredential);
+    }
+
+    private <C extends CdpCredential> C storeCdpCredential(C credential, Long stackId, BiConsumer<Stack, String> func) {
+        try {
+            transactionService.required(() -> {
+                String jsonString = new Json(credential).getValue();
+                Stack stackToSave = stackService.getStackById(stackId);
+                func.accept(stackToSave, jsonString);
+                stackService.save(stackToSave);
+            });
+            return credential;
+        } catch (TransactionService.TransactionExecutionException e) {
+            LOGGER.warn("Error during saving cdp credential.", e);
+            return null;
+        }
```

Comment:
shouldn't this fail the freeipa provision instead of ignoring the issue?

Reply:
no - at least not here, it could be valid to just skip the feature usage if it's not filled. and throw exception only if the credential missing for a critical feature, but those errors happens before/during pillar generation based on these values (so technically right after these operations)

Reply:
so it's ok if the json fails, or the stack save fails? at least I don't see any other case this could fail.

Reply:
if test fails the missing data should be handled later (probably will need to include hardcore failure for things like metering)